lambdas/service/app.py

```diff
@@ -957,57 +962,101 @@ def generate_manifest(event, context):
 
 
-@app.route('/dss/files/{uuid}', methods=['GET'], cors=True)
+dss_files_spec = {
+    'tags': ['DSS'],
+    'parameters': [
+        params.path('uuid', str, description='UUID of the file to be checked out'),
+        params.query(
+            'fileName',
+            schema.optional(str),
+            description='The desired name of the file. If absent, the UUID of the file will be used.'
+        ),
+        params.query(
+            'requestIndex',
+            schema.optional(int),
+            description='Number of attempts made through the endpoint to fetch the desired file.'
+        ),
+        params.query(
+            'wait',
+            schema.optional(int),
+            description=collapse_lines(
+                '''
+                If this parameter is 1 and the checkout is still in progress,
+                the server will wait before returning a response. This parameter
+                should only be set to 1 by clients who don't honor the
+                `Retry-After` header, preventing them from quickly exhausting
+                the maximum number of redirects. If the server cannot wait the
+                full amount, any amount of wait time left will still be returned
+                in the Retry-After header of the response.
+                '''
+            )
+        ),
+        params.query(
+            'replica',
+            schema.enum('aws'),
+            description=collapse_lines(
+                '''
+                All query parameters not mentioned above are forwarded to DSS in
+                order to initiate the checkout process for the correct file. For
+                more information refer to https://dss.data.humancellatlas.org
+                under `GET /files/{uuid}`. The only mandatory forwarded
+                parameter is `replica`, for which Azul only supports AWS.
+                '''
+            )
+        )
+    ]
+}
+
+
+@app.route('/dss/files/{uuid}', methods=['GET'], cors=True, method_spec={
+    **dss_files_spec,
+    'summary': 'Request a download link to a Data Store file and redirect',
+    'responses': {
+        '301': {
+            'description': collapse_lines(
+                '''
+                DSS checkout has been started or is ongoing. The response is a
+                redirect to this very endpoint, so the client should expect a
+                subsequent response of the same kind.
+                '''
+            ),
+            'headers': {
+                'Location': responses.header(str, description='This endpoint.'),
+                'Retry-After': responses.header(int, description=collapse_lines(
+                    '''
+                    Recommended number of seconds to wait before requesting the
+                    URL specified in the `Location` header. The response MAY
+                    carry this header even if server-side waiting was requested
+                    via `wait=1`.
+                    '''
+                ))
```

Comment:
```suggestion
                'Retry-After': responses.header(int, description=unwrap('''
                    Recommended number of seconds to wait before requesting the
                    URL specified in the `Location` header. The response MAY
                    carry this header even if server-side waiting was requested
                    via `wait=1`.
                '''))
```

unless `make format` mangles this

Reply:
Or even

```suggestion
                'Retry-After': responses.header(int, description=unwrap('''
                    Recommended number of seconds to wait before requesting the
                    URL specified in the `Location` header. The response MAY
                    carry this header even if server-side waiting was requested
                    via `wait=1`.'''))
```

Reply:
I'm also wondering if we should build the call to unwrap into the helpers. If a keyword argument named `description` is passed, apply `unwrap()` to it automatically.

Reply:
I think it's a good idea to implement this.